trackpaypal.php

```diff
@@ -216,4 +216,58 @@ function trackpaypal_civicrm_postIPNProcess(&$IPNData) {
   // and transaction details to send to Google Analytics
   // via its REST interface
   watchdog('paypal', 'IPN payload: %payload', array('%payload' => print_r($IPNData, TRUE)), WATCHDOG_DEBUG);
+
+  // Retrieve extension settings
+    $event_type = Civi::settings->get('trackpaypal_event_type');
+    $tracking_code = Civi::settings->get('trackpaypal_tracking_code');
+
+  // Check the GA Code is of valid syntax
+  // If not we do nothing
+  if (!trackpaypal_isGACode($tracking_code)) {
+    watchdog('paypal', 'GA Code invalid: %code ', array('%code' => $tracking_code), WATCHDOG_DEBUG);
```

Comment:
This is ok for Phase 1 but i think for Phase 2 looking at logging as per https://civicrm.org/blog/totten/the-static-is-dead-long-live-the-static might be better (more non CMS linked)

Reply:
Completely agree. `Civi::log()->error('GA Code invalid: {code}', array('code' => $tracking_code));` definitely looks smarter. Easy enough to update now I guess.